src/scripts/summary.js

```diff
@@ -410,17 +410,29 @@ class Summary extends H5P.EventDispatcher {
     for (const section of sections) {
       const sectionRow = document.createElement("li");
       sectionRow.classList.add('h5p-interactive-book-summary-overview-section-details');
+      const sectionRowContainer = document.createElement("div");
+      sectionRowContainer.classList.add('h5p-interactive-book-summary-overview-section-container');
+      sectionRow.appendChild(sectionRowContainer);
       if (this.behaviour.progressIndicators) {
         const icon = document.createElement("span");
         icon.classList.add('h5p-interactive-book-summary-section-icon');
         icon.classList.add(section.taskDone ? 'icon-chapter-done' : 'icon-chapter-blank');
-        sectionRow.appendChild(icon);
+        sectionRowContainer.appendChild(icon);
       }
 
       const title = document.createElement("button");
       title.type = "button";
       title.classList.add('h5p-interactive-book-summary-section-title');
       title.onclick = () => {
+        if (section.instance.libraryInfo.machineName === 'H5P.ActiveReaderTextInput') {
+          const contentChapter = document.querySelector(`.h5p-content-chapter-${chapterId}`);
```

Comment:
Please do not use query selectors if they are not absolutely necessary. They are slow and prone to error (and the result can in fact be `null` here if you have not interacted with the CKEditor before clicking).

You have full access to the Active reader, so please retrieve the DOM from the appropriate class instance.

Reply:
I think it is necessary, I am creating the div element on the fly for the content added in the active-reader-text-input, which does not have method to retrieve DOM. https://github.com/NDLANO/h5p-active-reader/pull/1/files#diff-44865373b09e58b12e9044bddc3a7b033ff7247ba5c3c1578d127800c87d9a3dR472

Reply:
I didn't say everything was there already. The `responseField` elements are created by the Active Reader (not by some other component with code out of our control) and can easily stored (e.g. in an object indexed by the `subContentId` or whatever is suitable) in order to retrieve them when needed here.

Reply:
This could be a way of doing or achieving the same result, I prefer doing this way and it is aligned with the existing libraries' code.

Reply:
The reason is, even if we use the element from the active-reader-input-text library we still need to put that into DOM and in a way there will be the same outcome.

Reply:
Ahh, I thought you were talking about the response method from the h5p-active-reader-text-input library. No worries we'll use the variable

Reply:
I think this is also covered. Please have a look.